src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
```

Comment:
Why ? Action type has restricted values (addProperties, clear,removeProperties), so let's use switch for them